accord-core/src/main/java/accord/messages/CheckStatus.java

```diff
@@ -409,22 +471,57 @@ public Known sufficientFor(Participants<?> participants, WithQuorum withQuorum)
             return sufficientFor(participants, saveStatus, route, partialTxn, committedDeps, writes, result);
         }
 
+        // it is assumed that we are invoking this for a transaction that will execute;
+        // the result may be erroneous if the transaction is invalidated, as logically this can apply to all ranges
+        public Ranges sufficientFor(Known required, Ranges expect)
+        {
+            Invariants.checkState(saveStatus != SaveStatus.Invalidated);
+            Known have = saveStatus.known;
+            if (!required.isSatisfiedBy(have))
+                return Ranges.EMPTY;
+
+            Ranges result = expect;
+            if (required.deps.hasDecidedDeps())
+                result = result.slice(committedDeps.covering, Minimal);
+
+            if (required.definition.isKnown())
+                result = result.slice(partialTxn.covering(), Minimal);
+
+            return result;
+        }
+
         private static Known sufficientFor(Participants<?> participants, SaveStatus saveStatus, Route<?> route, PartialTxn partialTxn, PartialDeps committedDeps, Writes writes, Result result)
         {
-            Status.Definition definition = saveStatus.known.definition;
+            Known known = saveStatus.known;
+            Status.KnownRoute knownRoute;
```

Comment:
We only don't currently send the `FullRoute` for `Accept`. So we require a `FullRoute` to transition to every other state. We could send it for `Accept` also, but since `Accept` is permitted to also not know the transaction it seems fine to permit it to only be a partial route. We also permit `NotDefined` to accept a random `Route` that may not even cover the current command store, so that it has some information with which to coordinate the transaction's invalidation - this happens only via `InformHomeOfTxn` where a replica needs to hand-off to a coordinating shard to know it can forget the transaction from its progress log.